github.js

```diff
@@ -292,14 +332,13 @@ GithubLocation.prototype = {
 
     // request the repo to check that it isn't a redirect
     return new Promise(function(resolve, reject) {
-      request({
+      request(extend({}, self.defaultRequestOptions, {
```

Comment:
Was thinking more along the lines of `extend({ uri: remoteString + repo, ... }, self.defaultRequestOptions)`. I know that will overwrite rather, but in this case that should be fine. Don't mean to be difficult honest - just trying to optimize for code readability.